pkgs/tools/security/hash-identifier/default.nix

```diff
@@ -0,0 +1,31 @@
+{ lib, fetchFromGitLab, python3, python3Packages }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "hash-identifier";
+  version = "1.2";
+
+  src = fetchFromGitLab {
+    owner = "kalilinux";
+    repo = "packages/hash-identifier";
+    rev = "20ebdd2f33ffc1bdcc010ff582be203696b0bb2e";
+    sha256 = "2cGF7lRXG+w7QsztbEWanvifPtjOwPNMvm660z4rxZA=";
+  };
+
+  # no setup.py is available
+  dontBuild = true;
+  doCheck = false;
+
+  installPhase = ''
+    mkdir -p ${placeholder "out"}/bin
+    cp hash-id.py ${placeholder "out"}/bin/hash-identifier
+    chmod +x ${placeholder "out"}/bin/hash-identifier
+  '';
```

Comment:
`buildPythonPackage` is usually for exporting python modules which can be imported. If this is a just a python script. you could probably just run `patchShebangs` on it, and call it good.

Reply:
Okay, so basically move to stdenv.mkDerivation and run patchShebangs?

Reply:
judging from https://gitlab.com/kalilinux/packages/hash-identifier/-/blob/kali/master/hash-id.py yes

Reply:
```nix
  buildInputs = [ python3 ];

  installPhase = ''
    patchShebangs ./hash-id.py
    mkdir -p ${placeholder "out"}/bin
    cp hash-id.py ${placeholder "out"}/bin/hash-identifier
    chmod +x ${placeholder "out"}/bin/hash-identifier
  '';
``` Clearly I'm doing this wrong, I'm just not sure why.

Reply:
probably want :

```
installPhase = ''
  patchShebangs --host ./hash-id.py
  mv ./hash-id.py hash-identifier
  install -m755 -t $out/bin hash-identifier
```

One concern I have though about renaming the script is, is that expected behavior?

Reply:
I'm actually not sure. Because in the README they just execute using python3 ./hash-id.py, but the package itself is called hash-identifier. I'm in favor of the rename, but maybe someone more experienced should weigh in? @fabaff

Reply:
Hm, the patchShebangs doesn't work. Odd. I've never used patchShebangs before so I'm a bit lost.

Reply:
About the naming. From my point of view we should use `hash-identifier`.